src/Kernel/BTrees.wl

```diff
@@ -196,6 +288,19 @@ SetAttributes[BTreeGamma, Listable];
 BTreeSigma[(BTree | BTreeN | BTreeDAE)[t_, ___]] := treeSigma[t];
 SetAttributes[BTreeSigma, Listable];
 
+BTreePrune[h_[t_,p__],h_[t_,p__]]/;(AnyTrue[{BTree,BTreeN,BTreeDAE},h===#&]):=h[\[FormalY],p];
```

Comment:
We check for the 3 types of trees so much, it would help to have the pattern `BTreeAny = _BTree | _BTreeN | _BTreeDAE;` as a private variable to reuse.  I think a public function `BTreeQ` would be handy as well